src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -1320,8 +1359,10 @@ subroutine gsi_fv3ncdf_readuv(dynvarsfile,ges_u,ges_v)
              iret=nf90_get_var(gfile_loc,k,uu)
              if(trim(name)=='u'.or.trim(name)=='U') then
                 allocate(temp1(dim(dim_id(1)),dim(dim_id(2)),dim(dim_id(3))))
+                if(.not.grid_reverse_flag) call reverse_grid_r_uv(uu,dim(dim_id(1)),dim(dim_id(2)),dim(dim_id(3)))
```

Comment:
This is confusing. if not grid reverse then call reverse grid??? 
I suggest that we add comments or reverse the logic.

Reply:
Thanks for the comments.